File: ray-tracer/scene/Ray.cs
using ray_tracer.common;

namespace ray_tracer.scene
{
    public class Ray
    {
        public readonly Vector origin;
        public readonly Vector direction;

        public Ray(Vector origin, Vector direction)
        {
            this.origin = origin;
            this.direction = direction;
        }
    }
}

File: java-app/src/pl/mikolajczuk/kamil/raytracing/lib/scene/Ray.java
package pl.mikolajczuk.kamil.raytracing.lib.scene;

import pl.mikolajczuk.kamil.raytracing.lib.common.Vector;

public class Ray {
    public final Vector origin;
    public final Vector direction;

    public Ray(Vector origin, Vector direction) {
        this.origin = origin;
        this.direction = direction;
    }
}
